### PR TITLE
Send email in English if current language don't have email template

### PR DESCRIPTION
**Pull request by martinfojtik**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Send email in English if current language or default language don't have email template. |
| Type? | improvement |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? |  http://forge.prestashop.com/browse/BOOM-1832
| How to test? | You can test it if you send email through contact form: |
|  | \- delete `mail\fr\contact.html` or `mails\fr\contact_form.html` |
|  | \- submit Contact form |
|  | \- check if you receive email in english |

### CODE REVIEW

**Comment by aleeks** [comment]
ping @vincentbz for the idea. Need to see with you.

File: classes/Mail.php [hunk 1]
```diff
@@ -310,19 +328,28 @@ public static function Send(
...
                 true
             );
-            if ($overrideMail && file_exists($templatePath.$iso.'/lang.php')) {
-                include_once($templatePath.$iso.'/lang.php');
-            } elseif ($moduleName && file_exists($themePath.'mails/'.$iso.'/lang.php')) {
-                include_once($themePath.'mails/'.$iso.'/lang.php');
-            } elseif (file_exists(_PS_MAIL_DIR_.$iso.'/lang.php')) {
-                include_once(_PS_MAIL_DIR_.$iso.'/lang.php');
+
+            foreach ($isoArray as $isoCode) {
```
Comment (julienbourdeau) [inline classes/Mail.php]: Are you sure you shouldn't break the loop when you find the first file?

File: classes/Mail.php [hunk 2]
```diff
@@ -258,37 +258,55 @@ public static function Send(
             $swift = \Swift_Mailer::newInstance($connection);
             /* Get templates content */
             $iso = Language::getIsoById((int) $idLang);
-            if (!$iso) {
+            $isoDefault = Language::getIsoById((int) Configuration::get('PS_LANG_DEFAULT'));
+            if (!$iso && !$isoDefault) {
                 Tools::dieOrLog(Tools::displayError('Error - No ISO code for email'), $die);
 
                 return false;
```
Comment (julienbourdeau) [inline classes/Mail.php]: Why not continuing with `en`?

**Review by julienbourdeau — DISMISSED** [review]

**Comment by julienbourdeau** [comment]
The file `lang.php` doesn't exist anymore, you can probably remove this code.

**Comment by dSkrbic** [comment]
Test Q&A **OK**

File: classes/Mail.php (hunk 2 above)
Comment (martinfojtik) [inline classes/Mail.php]: good idea, there is no need for this error message

**Comment by martinfojtik** [comment]
@julienbourdeau The `lang.php` still exist in `mail` folder, so i deleted it

**Comment by aleeks** [comment]
@julienbourdeau it's ok for you this PR ?

**Comment by aleeks** [comment]
Thank you @martinfojtik

**Pull request merged**